dockerfile_parse/parser.py

```diff
@@ -549,3 +532,28 @@ def _add_instruction(self, instruction, value):
             lines = self.lines
             lines += new_line
             self.lines = lines
+
+    @property
+    def context_structure(self):
+        """
+        :return: list of context objects
+            (Contains info about labels and environment variables for each line.)
+        """
+        instructions = []
+        envs = {}
+        labels = {}
```

Comment:
Yes, I like this better,  and having `context.update_line_value` behave that way makes sense to me.